modules/ximgproc/include/opencv2/ximgproc/edge_drawing.hpp

```diff
@@ -0,0 +1,404 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef __OPENCV_EDGE_DRAWING_HPP__
+#define __OPENCV_EDGE_DRAWING_HPP__
+
+#include <opencv2/core.hpp>
+
+// Circular arc, circle thresholds
+#define VERY_SHORT_ARC_ERROR     0.40  // Used for very short arcs (>= CANDIDATE_CIRCLE_RATIO1 && < CANDIDATE_CIRCLE_RATIO2)
+#define SHORT_ARC_ERROR          1.00  // Used for short arcs (>= CANDIDATE_CIRCLE_RATIO2 && < HALF_CIRCLE_RATIO)
+#define HALF_ARC_ERROR           1.25  // Used for arcs with length (>=HALF_CIRCLE_RATIO && < FULL_CIRCLE_RATIO)
+#define LONG_ARC_ERROR           1.50  // Used for long arcs (>= FULL_CIRCLE_RATIO)
```

Comment:
Defines should be avoided in public headers:
- look like they are not used in Public API, move to `src/` files
- `CV_` prefix is necessary to avoid conflicts
- prefer enums or global static constants (in classes) instead.